metricbeat/cmd/test/modules.go

```diff
@@ -0,0 +1,74 @@
+package test
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/testing"
+	"github.com/elastic/beats/metricbeat/beater"
+)
+
+func GenTestModulesCmd(name, beatVersion string) *cobra.Command {
+	return &cobra.Command{
+		Use:   "modules [module] [metricset]",
+		Short: "Test modules settings",
+		Run: func(cmd *cobra.Command, args []string) {
+			var filter_module, filter_metricset string
+			if len(args) > 0 {
+				filter_module = args[0]
+			}
+
+			if len(args) > 1 {
+				filter_metricset = args[1]
+			}
+
+			b, err := beat.New(name, beatVersion)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Error initializing beat: %s\n", err)
+				os.Exit(1)
+			}
+
+			err = b.Init()
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Error initializing beat: %s\n", err)
+				os.Exit(1)
+			}
+
+			config, err := b.BeatConfig()
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Error initializing beat: %s\n", err)
```

Comment:
The three "Error initializing beat" message above might get confusing at some point.